include/aws/common/byte_buf.h

```diff
@@ -450,10 +450,11 @@ AWS_EXTERN_C_END
  */
 AWS_STATIC_IMPL struct aws_byte_buf aws_byte_buf_from_c_str(const char *c_str) {
     struct aws_byte_buf buf;
-    buf.buffer = (uint8_t *)c_str;
-    buf.len = strlen(c_str);
+    buf.len = (!c_str) ? 0 : strlen(c_str);
     buf.capacity = buf.len;
+    buf.buffer = (buf.capacity == 0) ? NULL : (uint8_t *)c_str;
```

Comment:
`buf.buffer = (uint8_t *)c_str` should be fine, `NULL` casted to `uint8_t *` is still `NULL` 😛

Reply:
Oh I see you're branching on size, not pointer. I don't love this, but I guess it is at least consistent with from_array.